includes/thesisdefs.tex

```diff
@@ -32,55 +13,58 @@
 \newcommand\@thesistitlemedskip{0.2in}
 \newcommand\@thesistitlebigskip{0.6in}
 \newcommand{\degree}[1]{\gdef\@degree{#1}}
-\newcommand{\project}{\gdef\@doctype{A masters project report}}
+\newcommand{\doctype}[1]{\gdef\@doctype{#1}}
+\newcommand{\masters}{\gdef\@doctype{Masters project report}}
 \newcommand{\prelim}{\gdef\@doctype{A preliminary report}}
-\newcommand{\thesis}{\gdef\@doctype{A thesis}}
-\newcommand{\dissertation}{\gdef\@doctype{A dissertation}}
-\newcommand{\department}[1]{\gdef\@department{(#1)}}
+\newcommand{\dissertation}{\gdef\@doctype{Dissertation}}
+\newcommand{\department}[1]{\gdef\@department{#1}}
+\newcommand{\defensedate}[1]{\gdef\@defensedate{#1}}
+\newcommand{\advisorname}[1]{\gdef\@advisorname{#1}}
+\newcommand{\advisortitle}[1]{\gdef\@advisortitle{#1}}
+\newcommand{\advisoraffiliation}[1]{\gdef\@advisoraffiliation{#1}}
+
+\def\@committee{\@empty}
+\newcommand{\addcommittee}[3]{
+    \edef\@committee{\@committee\if\@committee\@empty\else\ \fi  \tabularnewline #1, #2, #3}
+}
+\newcommand{\showcommittee}{\begin{tabular}[t]{l}
+  \@advisorname, \@advisortitle, \@advisoraffiliation
+  \@committee
+\end{tabular}}
+
 
 \newenvironment{titlepage}
  {\@restonecolfalse\if@twocolumn\@restonecoltrue\onecolumn
   \else \newpage \fi \thispagestyle{empty}
 % \c@page\z@ -- deleted: count title page in thesis
 }{\if@restonecol\twocolumn \else \newpage \fi}
 
-\gdef\@degree{Doctor of Philosophy}    %Default is PhD
-\gdef\@doctype{Dissertation}         %Default is dissertation, use "A Preliminary Report" for prelim
-
-\gdef\@department{(Engineering Physics)} 
-\gdef\@defensedate{2025-05-01}
-\gdef\@committee{
-  Paul P.H. Wilson, Professor, Engineering Physics\\
-  Committee Member 2\\
-  Another Committee Member\\
-  And yet another committee member name and title\\
-  }
-
 \renewcommand{\maketitle}{%
   \begin{titlepage}
-%-----------------------------------------------------------------------------
-% -- The thesis office doesn't like thanks on title page.  Put it in
-% -- the acknowledgments.  This is here so you don't have to change
-% -- your titlepage when converting from report style. -> from Purdue, but I
-%        left it here since it seems compatible with UW-Madison, Eric
-%-----------------------------------------------------------------------------
-    \def\thanks##1{\typeout{Warning: `thanks' deleted from thesis titlepage.}}
+    \begin{OnehalfSpace}
+
     \let\footnotesize\small \let\footnoterule\relax \setcounter{page}{1}
     \begin{center}
       {\textbf{\expandafter\expandafter{\@title}}} \\[\@thesistitlemedskip]
        by \\[\@thesistitlemedskip]
+
       \@author \\[\@thesistitlebigskip]
+
       \@doctype\ submitted in partial fulfillment of \\
       the requirements for the degree of\\[\@thesistitlemedskip]
       \@degree \\[\@thesistitlemedskip]
```

Comment:
Add a definition of the program (e.g. Nuclear Engineering and Engineering Physics) wherever the dept and title, etc, are defined
```suggestion
      \@degree \\
      in \\
      \@program \\[\@thesistitlemedskip]
```